src/lib/EDSL.mli

```diff
@@ -288,7 +288,58 @@ val tmp_file: ?tmp_dir: string t -> string -> file
 
 (** {3 Command Line Parsing} *)
 
+(** Typed command-line parsing for your shell scripts, à la {!Prtinf.scanf}. *)
 module Command_line: sig
+  (** 
+
+     Use this module like OCaml's {!Printf.scanf} function.
+
+     - Build a command-line “format specification” using the {!Arg} module.
+     - Call the {!parse} function with an appropriately typed function.
+
+     Example: {[
+       let cli_spec =
+         Command_line.Arg.(
+           string
+             ~doc:"The URL to the stuff" ["-u"; "--url"]
+             ~default:no_value
+           & flag ["-c"; "--all-in-tmp"] ~doc:"Do everything in the temp-dir"
+           & string ["-f"; "--local-filename"]
+             ~doc:"Override the downloaded file-name"
+             ~default:no_value
+           & string ["-t"; "--tmp-dir"]
+             ~doc:"Use <dir> as temp-dir"
+             ~default:(Genspio.EDSL.string "/tmp/genspio-downloader-tmpdir")
+           & usage "Download archives and decrypt/unarchive them.\n\
+                    ./downloader -u URL [-c] [-f <file>] [-t <tmpdir>]"
+         ) in
+       (*
+          `cli_spec` has type:
+
+           (string Genspio.EDSL.t ->
+            bool Genspio.EDSL.t ->
+            string Genspio.EDSL.t -> string Genspio.EDSL.t -> unit Genspio.EDSL.t,
+            unit Genspio.EDSL.t)
+           Genspio.EDSL.Command_line.cli_options
+          
+           so the second argument must have type:
```

Comment:
Change to, " so the `action` function (the second argument to `parse`) must have type."